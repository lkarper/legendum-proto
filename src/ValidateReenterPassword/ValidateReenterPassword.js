import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateReenterPassword = (props) => {
    const { 
        password,
        reenterPassword,
        reenterPasswordError,
        setReenterPasswordError,
    } = props;

    useEffect(() => {
        if (password === reenterPassword) {
            setReenterPasswordError('');
        } else {
            setReenterPasswordError('Passwords do not match.');
        }
    }, [password, reenterPassword, reenterPasswordError, setReenterPasswordError]);

    const validationMessage = reenterPasswordError
        ? 
            <p 
                id='reenter-password-error'
                style={{ color: 'darkred' }}
            >
                {reenterPasswordError}
            </p>
        :
            <p id='reenter-password-error'>Passwords match.</p>;

    return (
        <div role='alert'>
            {validationMessage}
        </div>
    );
}

ValidateReenterPassword.defaultProps = {
    password: '', 
    reenterPassword: '', 
    reenterPasswordError: '',
    setReenterPasswordError: () => {},
};

ValidateReenterPassword.propTypes = {
    password: PropTypes.string.isRequired, 
    reenterPassword: PropTypes.string.isRequired, 
    reenterPasswordError: PropTypes.string.isRequired,
    setReenterPasswordError: PropTypes.func.isRequired,
};

export default ValidateReenterPassword;
