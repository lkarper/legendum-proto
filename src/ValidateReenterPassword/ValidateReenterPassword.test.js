import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateReenterPassword from './ValidateReenterPassword';

describe('ValidateReenterPassword component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateReenterPassword />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateReenterPassword />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
