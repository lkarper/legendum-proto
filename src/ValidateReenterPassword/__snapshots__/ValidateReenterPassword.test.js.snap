// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateReenterPassword component renders the UI as expected 1`] = `
<ValidateReenterPassword
  password=""
  reenterPassword=""
  reenterPasswordError=""
  setReenterPasswordError={[Function]}
>
  <div
    role="alert"
  >
    <p
      id="reenter-password-error"
    >
      Passwords match.
    </p>
  </div>
</ValidateReenterPassword>
`;
